python_jsonschema_objects/validators.py

```diff
@@ -95,13 +108,13 @@ def check_boolean_type(param, value, _):
 
 @type_registry.register(name='integer')
 def check_integer_type(param, value, _):
-    if not isinstance(value, int):
+    if not isinstance(value, int) or isinstance(value, bool):
```

Comment:
Is there a reason you changed this to allow booleans for integer and number values?